src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -151,11 +159,29 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (school == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, School.class.getSimpleName()));
+        }
+        if (!school.isEmpty() && !School.isValidSchName(school)) {
```

Comment:
check `isEmpty` inside `isValidSchName`
```suggestion
        if (!School.isValidSchName(school)) {
```